src/orng.cpp

```diff
@@ -926,5 +1111,57 @@ bool orng::_deliver_random(eosio::name dapp, uint64_t assoc_id, const eosio::che
 
     return use_legacy;
 }
+uint64_t orng::_fetch_wax_usd_price() {
+    // Get the waxpusd pair info from Delphioracle
+    delphioracle::pairs_t pairs_table(delphioracle::DELPHIORACLE_ACCOUNT, delphioracle::DELPHIORACLE_ACCOUNT.value);
+    auto pair_it = pairs_table.require_find("waxpusd"_n.value, "waxpusd pair not found");
+
+    // Get quoted precision
+    uint64_t quoted_precision = pair_it->quoted_precision;
+
+    // Get the last datapoint (highest ID) from datapoints table scoped by waxpusd
+    delphioracle::datapoints_t datapoints_table = delphioracle::get_datapoints("waxpusd"_n);
+    check(datapoints_table.begin() != datapoints_table.end(), "no datapoints found for waxpusd");
+
+    // Get the last entry (highest id)
+    auto last_it = datapoints_table.end();
+    --last_it;
+
```

Comment:
is it possible to check that the price entry is not too stale with something like:
```
check(current_time_point() - last_it->timestamp < eosio::seconds(3600), "price data too stale");
```